sofar/sofastream.py

```diff
@@ -0,0 +1,192 @@
+from netCDF4 import Dataset
+import numpy as np
+
+class SofaStream():
+    """
+    Get specific data from SOFA-file without reading entire file into memory.
+
+    :class:`SofaStream` opens a SOFA-file and retreives only the requested
+    data. To access certain data just append its name to your
+    :class:`SofaStream`-instance, e.g. ``'.GLOBAL_RoomType'``.
+    :class:`SofaStream` uses the namespace of sofar – for more information on
+    available data and its associated names refer to the :ref:`conventions
+    section <conventions_introduction>`. Also make sure to check the examples
+    below on how to use :class:`SofaStream`.
+
+    Parameters
+    ----------
+    filename : str
+        Full path to a SOFA-file
+
+    Returns
+    --------
+    sofastream : SofaStream
+
+    Notes
+    -----
+    Accessing data:
+
+    SofaStream is supposed to be used within a with-statement. To acces data
+    append the corresponding name to your instance of :class:`SofaStream`.
+
+    Depending on the type of the requested data you either get the values
+    directly or need to slice the returned variable:
+
+    - If an attribute is called, its value is returned directly.
+    - If a variable is called, a netCDF4-variable is returned. To get the
+      values it needs to be sliced (see examples).
+
+    Examples
+    --------
+    Get an attribute from a SOFA-file:
+
+        >>> from sofar import SofaStream
+        >>> filename = "path/to/file.sofa"
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.GLOBAL_RoomType
+        >>>     print(data)
+        free field
+
+    Get a variable from a SOFA-file:
+
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.Data_IR
+        >>>     print(data)
+        <class 'netCDF4._netCDF4.Variable'>
+        float64 Data.IR(M, R, N)
+        unlimited dimensions:
+        current shape = (11950, 2, 256)
+        filling on, default _FillValue of 9.969209968386869e+36 used
+
+    What is returned is a `netCDF-variable`. To access the values (in this
+    example the IRs) the variable needs to be sliced:
```

Comment:
we can add here, that if you want to use all the data you should go for the Sofa class instead